### PR TITLE
Add API documentation for upsampling operator with examples

### PR DESCRIPTION
**Pull request by sandeep-krishnamurthy**
## Description ##
Fixes https://github.com/apache/incubator-mxnet/issues/12970
Upsampling operator did not had valid documentation and example usage leading to issues like - #12970

### Essentials ###
Please feel free to remove inapplicable items for your PR.

- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- Add clear API documentation for upsampling operator

@aaronmarkham @vandanavk  - Can you please help review this?

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [pr-awaiting-review, Doc]

@sandeep-krishnamurthy LGTM. Could you add a link to the preview of the webpage once CI passes?

**Comment by sandeep-krishnamurthy** [comment]
http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-14919/1/api/python/ndarray/ndarray.html?highlight=upsampling#mxnet.ndarray.UpSampling

File: src/operator/nn/upsampling.cc [hunk 1]
```diff
@@ -121,9 +121,67 @@ struct UpSamplingGrad {
 DMLC_REGISTER_PARAMETER(UpSamplingParam);
 
 NNVM_REGISTER_OP(UpSampling)
-.describe("Performs nearest neighbor/bilinear up sampling to inputs. "
-          "Bilinear upsampling makes use of deconvolution. Therefore, "
-          "provide 2 inputs - data and weight. ")
+.describe(R"code(Upsamples the given input data.
+
+2 algorithms (``sample_type``) are available for upsampling
```
Comment (aaronmarkham) [inline src/operator/nn/upsampling.cc]:
```suggestion
Two algorithms (``sample_type``) are available for upsampling:
```

File: src/operator/nn/upsampling.cc [hunk 2]
```diff
@@ -121,9 +121,67 @@ struct UpSamplingGrad {
...
+     [1. 1. 1. 1. 1. 1.]
+     [1. 1. 1. 1. 1. 1.]
+     [1. 1. 1. 1. 1. 1.]]]]
+  <NDArray 1x1x6x6 @cpu(0)>
+
+**Bilinear Upsampling**
+
+Uses `deconvolution` algorithm under the hood. You need provide both input data and the kernel.
+
+Input data is expected to be 4D of the form `(batch, channel, height, width)`.
```
Comment (aaronmarkham) [inline src/operator/nn/upsampling.cc]: Why not just say:
```suggestion
Input data is expected to be NCHW.
```

File: src/operator/nn/upsampling.cc [hunk 3]
```diff
@@ -121,9 +121,67 @@ struct UpSamplingGrad {
...
+.describe(R"code(Upsamples the given input data.
+
+2 algorithms (``sample_type``) are available for upsampling
+
+- Nearest Neighbor
+- Bilinear
+
+**Nearest Neighbor Upsampling**
+
+Input data is expected to be 4D of the form `(batch, channel, height, width)`.
```
Comment (aaronmarkham) [inline src/operator/nn/upsampling.cc]: Why not just say:
```suggestion
Input data is expected to be NCHW.
```

**Comment by sandeep-krishnamurthy** [comment]
> A couple of suggestions.
> Also, in another PR is was suggested by @eric-haibin-lin that the [Python examples be placed in a special docs generator class](https://github.com/apache/incubator-mxnet/pull/14243), and the C example be left as pseudo-code.

Makes sense, will make it pseudocode. Other comments addressed. Thanks.

**Comment by sandeep-krishnamurthy** [comment]
@aaronmarkham - Addressed the comments, made it pseudocode.

**Pull request merged**